src/course-unit/xblock-container-iframe/hooks/useMessageHandlers.tsx

```diff
@@ -20,16 +22,25 @@ export const useMessageHandlers = ({
   handleDuplicateXBlock,
   handleScrollToXBlock,
   handleManageXBlockAccess,
+  handleShowLegacyEditXBlockModal,
+  handleCloseLegacyEditorXBlockModal,
+  handleSaveEditedXBlockData,
+  handleFinishXBlockDragging,
 }: UseMessageHandlersTypes): MessageHandlersTypes => useMemo(() => ({
   [messageTypes.copyXBlock]: ({ usageId }) => dispatch(copyToClipboard(usageId)),
   [messageTypes.deleteXBlock]: ({ usageId }) => handleDeleteXBlock(usageId),
   [messageTypes.newXBlockEditor]: ({ blockType, usageId }) => navigate(`/course/${courseId}/editor/${blockType}/${usageId}`),
   [messageTypes.duplicateXBlock]: ({ blockType, usageId }) => handleDuplicateXBlock(blockType, usageId),
   [messageTypes.manageXBlockAccess]: ({ usageId }) => handleManageXBlockAccess(usageId),
-  [messageTypes.scrollToXBlock]: debounce(({ scrollOffset }) => handleScrollToXBlock(scrollOffset), 3000),
+  [messageTypes.scrollToXBlock]: debounce(({ scrollOffset }) => handleScrollToXBlock(scrollOffset), 1000),
   [messageTypes.toggleCourseXBlockDropdown]: ({
     courseXBlockDropdownHeight,
   }: { courseXBlockDropdownHeight: number }) => setIframeOffset(courseXBlockDropdownHeight),
+  [messageTypes.editXBlock]: ({ id }) => handleShowLegacyEditXBlockModal(id),
+  [messageTypes.closeXBlockEditorModal]: () => handleCloseLegacyEditorXBlockModal(),
+  [messageTypes.saveEditedXBlockData]: () => handleSaveEditedXBlockData(),
+  [messageTypes.studioAjaxError]: ({ error }) => handleResponseErrors(error, dispatch, updateSavingStatus),
+  [messageTypes.refreshPositions]: () => handleFinishXBlockDragging(),
```

Comment:
[optional] We can simplify it a bit:
```suggestion
  [messageTypes.closeXBlockEditorModal]: handleCloseLegacyEditorXBlockModal,
  [messageTypes.saveEditedXBlockData]: handleSaveEditedXBlockData,
  [messageTypes.studioAjaxError]: ({ error }) => handleResponseErrors(error, dispatch, updateSavingStatus),
  [messageTypes.refreshPositions]: handleFinishXBlockDragging,
```

Reply:
Corrected, thanks